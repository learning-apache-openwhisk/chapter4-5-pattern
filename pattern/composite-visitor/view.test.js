const main = require("./view").main

const Component = require("./lib/component")
const Visitor = require("./lib/visitor")

test("composite", () => {
    let v = new Visitor()
    let c = new Component("test")
    c.accept(v)
    expect(v.render()).toMatchSnapshot()
})

test("composite-visitor-view", () => {
    let name = "Mike"
    expect(main({name})).toMatchSnapshot()
})


test("composite render", () => {
	let name = new Component(`<input name="name">`)
	let email = new Component(`<input name="email">`)
	let phone = new Component(`<input name="phone">`)

	let form = new Component(`<form>`, `</form>`)
	form.add(name)
	form.add(email)
	form.add(phone)â€¨ 
	let page = new Component(`<html>`, `</html>`)
	page.add(form)

       let v = new Visitor()
       page.accept(v)
       console.log(v.prefixes)
       console.log(v.suffixes)
       console.log(v.render())

})
