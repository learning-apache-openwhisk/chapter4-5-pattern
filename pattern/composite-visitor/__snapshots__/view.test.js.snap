// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`composite 1`] = `"test"`;

exports[`composite-visitor-view 1`] = `
Object {
  "body": "<!DOCTYPE html>
  <html>
    <head>
     <link href=\\"//maxcdn.bootstrapcdn.com/bootstrap/3.3.0/css/bootstrap.min.css\\"
           rel=\\"stylesheet\\" id=\\"bootstrap-css\\">
    </head>
    <body>
     <div id=\\"container\\">
       <div class=\\"row\\">
         <div class=\\"col-md-8 col-md-offset-2\\">
          <h4><strong>Get in Touch</strong></h4>
  <form method=\\"POST\\">
    <div class=\\"form-group\\">
      <input type=\\"text\\" name=\\"name\\"
       value=\\"Mike\\"
       class=\\"form-control\\" placeholder=\\"Name\\">
    </div>
    <div class=\\"form-group\\">
     <input type=\\"text\\" name=\\"email\\"
      value=\\"\\"
      class=\\"form-control\\" placeholder=\\"Email\\">
    </div>
    <div class=\\"form-group\\">
      <input type=\\"tel\\" name=\\"phone\\"
          value=\\"\\"
          class=\\"form-control\\" placeholder=\\"Phone\\">
    </div>
    <button class=\\"btn btn-default\\" 
      type=\\"submit\\" name=\\"button\\">
       Send
     </button>
  </form>
         </div>
       </div>
     </div>
    </body>
  </html>",
}
`;
